api/envoy/config/listener/v3/listener.proto

```diff
@@ -88,6 +88,11 @@ message Listener {
     }
   }
 
+  // Configuration for envoy internal listener. All the future internal listener features should be added here.
+  // [#not-implemented-hide:]
+  message InternalListenerConfig {
+  }
```

Comment:
Is there really no configuration?

Reply:
Not yet... Too early to add. We can add later, right?

Reply:
You may want to add config fields along with the rest of the implementation.

Reply:
> You may want to add config fields along with the rest of the implementation.

Exactly. The empty `InternalListenerConfig` can enable the first version of the internal listener. And that is a sign to group internal listener specific field here